### PR TITLE
use of --fringe-samples=NOMINAL for cube2mseed and merge multiple fil…

### PR DESCRIPTION
**Pull request by jegestrich**
…es for one day with fill_value=0

Pull request to resolve issue #13 (Data gaps when converting to miniseed). See the response to the issue for details on the justification of certain changes. I included some comments on new code lines. The if statements to go through the multiple files for one day (if there is a data gap) can probably be optimized. Open for suggestions.

### CODE REVIEW

File: cube_convert.py [hunk 1]
```diff
@@ -200,10 +204,45 @@
 # Loop through each cut file and assign the channel number, editing the simple
 # metadata (automatically distinguish between a 3-element array or single
 # sensor)
-t_min, t_max = np.inf, -np.inf  # Initialize time bounds
+
+mdayhour = []
 for file in cut_file_list:
+    if file[-7:-6] == '.':
```
Comment (liamtoney) [inline cube_convert.py:210]: Can add a comment here explaining the significance of `file[-7:-6] == '.'` ?

File: cube_convert.py [hunk 2]
```diff
@@ -200,10 +204,45 @@
...
+    if mn < len(inds):
+        if nf == inds[mn][0]: # Check if the index of current hour matches a hour that has multiple files
+            print('found multiple files for one hour:')
+            st = obspy.read(file)
+            for i in range(len(inds[mn]) - 1):
+                print(os.path.basename(cut_file_list[inds[mn][i + 1]]))
+                st += obspy.read(cut_file_list[inds[mn][i + 1]])
+                os.remove(cut_file_list[inds[mn][i + 1]]) # Remove all files for that hour but the first one
+            while np.any([st[i].stats.sampling_rate != st[j].stats.sampling_rate for i in range(len(st)) for j in range(len(st))]): #check if all sampling rates are correct
+                st.remove(st[0]) #remove oldest trace (assuming that the sampling range was changed and the latest one is what we want)
```
Comment (liamtoney) [inline cube_convert.py:235]: Perhaps add a warning that we're removing a Trace here?

File: cube_convert.py [hunk 3]
```diff
@@ -200,10 +204,45 @@
...
+            st.merge(fill_value=0) # Fill data gaps with zero value
```
Comment (liamtoney) [inline cube_convert.py:236]: Maybe also a warning here that we're merging with `fill_value=0` so folks know to expect that...

File: cube_convert.py (hunk 2 above)
Comment (liamtoney) [reply cube_convert.py:235]: Especially since this is iteratively removing the oldest trace `st[0]`

**Review by liamtoney — CHANGES_REQUESTED** [review]
Thanks for this improvement! I made a couple comments on places to add more info / warnings — thanks for commenting it well already.

Have you tested on a case where there are no gaps, just to ensure that the old behavior is replicated?

File: cube_convert.py (hunk 1 above)
Comment (jegestrich) [reply cube_convert.py:210]: Yes I can. If there is a gap within one minute the mseedcut command produces two files with the same name but the second file will have a ".1" inserted before the ".pri0" (e.g. c0af0180628020000.pri0 and c0af0180628020000.1.pri0). The for loop and variable names `mdayhour` is a list of strings of the month, day and hour of each file. This is later used to determine if there are multiple files for each hour.

File: cube_convert.py (hunk 2 above)
Comment (jegestrich) [reply cube_convert.py:236]: Yes. I added a print statement

File: cube_convert.py (hunk 3 above)
Comment (jegestrich) [reply cube_convert.py:238]: Yes. I added a print statement.

**Review by liamtoney — APPROVED** [review]
This looks good to me now, as long as the standard behavior for non-gappy data is retained! I imagine @davidfee5 will want to take a look too before we merge.

**Comment by liamtoney** [comment]
@jegestrich wanna merge this one?

**Pull request merged**